src/server/controller/assessment/update.ts

```diff
@@ -0,0 +1,32 @@
+import { ActivityLogMessage, Assessment } from '@meta/assessment'
+import { User } from '@meta/user'
+
+import { BaseProtocol, DB } from '@server/db'
+import { AssessmentRepository } from '@server/repository/assessment/assessment'
+import { ActivityLogRepository } from '@server/repository/public/activityLog'
+
+export const update = async (
+  props: { user: User; assessment: Assessment },
+  client: BaseProtocol = DB
+): Promise<Assessment> => {
+  const { assessment, user } = props
+  // await AssessmentRepository.updateAssessmentSchema({ assessment }, client)
+
+  return client.tx(async (t) => {
+    const updatedAssessment = await AssessmentRepository.updateAssessment({ assessment }, t)
+
+    await ActivityLogRepository.insertActivityLog(
+      {
+        activityLog: {
+          target: updatedAssessment,
+          section: 'assessment',
+          message: ActivityLogMessage.assessmentUpdate,
+          user,
+        },
+        assessment: updatedAssessment,
```

Comment:
this should be `assessment` and not `updatedAssessment`

Reply:
Are you sure? 

In line 15 the const is defined as `updatedAssessment`. Do you mean to change the line 15 as well?

Reply:
Activity log is to keep track of changes
in the assessment we pass the 'old' assessment and the 'target' is the updated one

This way when you are looking at changes, it's easy to see what changed

Reply:
The repository method name must be renamed to update , that's what @sorja meant

Reply:
> The repository method name must be renamed to update , that's what @sorja meant

Thats also a good point though; what I mean is the activity log entries